VegProcessor/utils.py

```diff
@@ -325,6 +329,57 @@ def generate_pct_cover_custom(
     # da_out.to_netcdf("./pct_cover.nc")
     return da_out
 
+def calculate_buffered_land_use_percentages(
+    land_cover_da: xr.DataArray,
+    resolution_m: int = 60,
+    buffer_miles: float = 0.5
+) -> dict[str, xr.DataArray]:
+    """
+    Performs a focal analysis to calculate the percentage of broad land use
+    categories within a circular buffer around each pixel.
+    """
+    # Define the land use groups directly
+    land_use_groups = {
+        "nonhabitat": [2, 3, 4, 5],
+        "active_ag_water": [6, 13, 14, 24, 25, 26],
+        "pasture": [7, 8],
+        "forested": [9, 10, 11, 15, 16, 17, 18, 19, 20, 21, 22, 23],
+        "abandoned_ag": [12]
+    }
+
+    # Define the buffer kernel in pixel units
+    buffer_m = buffer_miles * 1609.34
+    radius_pixels = int(round(buffer_m / resolution_m))
+    kernel = disk(radius_pixels)
+
+    normalized_kernel = kernel.astype(np.float32) / np.sum(kernel)
+    
+    # Perform focal analysis for each land use group
+    percentage_arrays = {}
+    for name, codes_list in land_use_groups.items():
+        print(f"Calculating buffered percentages for: {name}...")
+        
+        # Create a binary mask where pixels matching the group's codes are True
+        binary_mask = np.isin(land_cover_da.values, codes_list).astype(np.float32)
+
+        # Calculate the mean over a footprint
+        focal_mean = convolve(
```

Comment:
Neat solution using `convolve` w/ the boolean mask here--pixel weight times 0 or 1 then sum everything to get a proportion. concise!